diff --git a/src/bz-entry.c b/src/bz-entry.c
index 65f0ff3..bbd967a 100644
--- a/src/bz-entry.c
+++ b/src/bz-entry.c
@@ -518,7 +518,8 @@ bz_entry_set_property (GObject      *object,
     case PROP_REMOTE_REPO_NAME:
       g_clear_pointer (&priv->remote_repo_name, g_free);
       priv->remote_repo_name = g_value_dup_string (value);
-      priv->is_flathub       = g_strcmp0 (priv->remote_repo_name, "flathub") == 0;
+      priv->is_flathub       = g_strcmp0 (priv->remote_repo_name, "flathub") == 0 ||
+                               g_strcmp0 (priv->remote_repo_name, "flathub-verified") == 0;
       g_object_notify_by_pspec (object, props[PROP_IS_FLATHUB]);
       break;
     case PROP_URL:
diff --git a/src/bz-flatpak-entry.c b/src/bz-flatpak-entry.c
index 9a74840..6f05df4 100644
--- a/src/bz-flatpak-entry.c
+++ b/src/bz-flatpak-entry.c
@@ -651,8 +651,9 @@ bz_flatpak_entry_new_for_ref (FlatpakRef    *ref,
         }
 
       share_urls = g_list_store_new (BZ_TYPE_URL);
-      if (kinds & BZ_ENTRY_KIND_APPLICATION &&
-          g_strcmp0 (remote_name, "flathub") == 0)
+      if (kinds & BZ_ENTRY_KIND_APPLICATION && (
+          g_strcmp0 (remote_name, "flathub") == 0 ||
+          g_strcmp0 (remote_name, "flathub-verified") == 0))
         {
           g_autofree char *flathub_url = NULL;
           g_autoptr (BzUrl) url        = NULL;
@@ -1029,7 +1030,9 @@ bz_flatpak_entry_new_for_ref (FlatpakRef    *ref,
         }
     }
 
-  if (component != NULL && g_strcmp0 (remote_name, "flathub") == 0)
+  if (component != NULL && (
+      g_strcmp0 (remote_name, "flathub") == 0 ||
+      g_strcmp0 (remote_name, "flathub-verified") == 0))
     {
       const char *verified_str     = NULL;
       const char *method           = NULL;
diff --git a/src/bz-flatpak-instance.c b/src/bz-flatpak-instance.c
index 163d323..ec160b9 100644
--- a/src/bz-flatpak-instance.c
+++ b/src/bz-flatpak-instance.c
@@ -687,7 +687,8 @@ check_has_flathub_fiber (CheckHasFlathubData *data)
         continue;
 
       name = flatpak_remote_get_name (remote);
-      if (g_strcmp0 (name, "flathub") == 0)
+      if (g_strcmp0 (name, "flathub") == 0 ||
+          g_strcmp0 (name, "flathub-verified") == 0)
         return dex_future_new_true ();
     }
   return dex_future_new_false ();
@@ -710,10 +711,10 @@ ensure_flathub_fiber (EnsureFlathubData *data)
 
   if (self->system != NULL)
     sys_remote = flatpak_installation_get_remote_by_name (
-        self->system, "flathub", cancellable, NULL);
+        self->system, "flathub-verified", cancellable, NULL);
   if (self->user != NULL)
     usr_remote = flatpak_installation_get_remote_by_name (
-        self->user, "flathub", cancellable, NULL);
+        self->user, "flathub-verified", cancellable, NULL);
 
   if (sys_remote != NULL)
     remote = g_steal_pointer (&sys_remote);
